contributors/design-proposals/testing/test-frameworks-repo.md

```diff
@@ -0,0 +1,81 @@
+# New git repo: k8s.io/testframeworks
+
+Status: Pending
+
+Version: N/A
+
+Implementation Owners: @totherme @hoegaarden and @apelisse
+
+## Motivation: Why do we need a new repo?
+
+To avoid an vendoring cycle.
+
+Some folks in Sig-CLI have been writing an [integration testing
+framework](https://github.com/kubernetes/kubectl/tree/master/pkg/framework/test)
+which they would like to use to simplify some of their integration tests which
+live in `k8s.io/kubernetes`. The framework currently lives in `k8s.io/kubectl`.
+Since `k8s.io/kubectl` already vendors components _from_ `k8s.io/kubernetes`, it 
+cannot be vendored _into_ `k8s.io/kubernetes`. 
+
+It therefore seems like a good idea to create a new repo to contain testing
+frameworks. The intent is that this repo could be vendored anywhere.
```

Comment:
"anywhere" includes k/k?

Reply:
If yes, this implies that it must not import anything from the staging repos, i.e. client-go, apimachinery, api, etc.

Reply:
discussed below.